flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasEpsilon.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.DoubleParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared epsilon param. */
+public interface HasEpsilon<T> extends WithParams<T> {
+
+    Param<Double> EPSILON =
+            new DoubleParam(
+                    "epsilon",
+                    "Convergence tolerance for iterative algorithms. The default value is 0.1",
+                    0.1,
```

Comment:
I have similar thoughts as @yunfengzhou-hub. Here are my findings that may be useful to consider here.

Spark and Scikit-learn [1] uses HasTol for this purpose. Logistic Regression wiki [2] mentions tolerance instead of epsilon. I searched on Google for words that are commonly used for determining the "termination criteria". It looks like tolerance is much more popular than epsilon in the machine learning domain (e.g. [3]).

[1] https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html
[2] https://en.wikipedia.org/wiki/Logistic_regression
[3] https://support.minitab.com/en-us/minitab/18/help-and-how-to/modeling-statistics/regression/how-to/nonlinear-regression/interpret-the-results/all-statistics-and-graphs/methods-and-starting-values/